src/main/java/interface_adapters/tui/controllers/RecipeNoteOperation.java

```diff
@@ -0,0 +1,171 @@
+import interface_adapters.tui.TextualOperation;
+import interface_adapters.tui.TextualReader;
+import usecase.RecipeManager;
+import usecase.RecipeNoteTaker;
+
+import java.util.List;
+
+
+/**
+ * An operation for running the note taker use-case.
+ */
+public class RecipeNoteOperation implements TextualOperation {
+    private final TextualReader reader;
+    private final RecipeNoteTaker recipeNoteTaker;
+    private final RecipeManager recipeManager;
+
+    private final List<TextualOperation> operations = List.of(
+            new NoteCreator(),
+            new NoteViewer(),
+            new NoteEditor(),
+            new NoteDeleter()
+    );
+
+    public RecipeNoteOperation(TextualReader reader, RecipeNoteTaker recipeNoteTaker, RecipeManager recipeManager) {
+        this.reader = reader;
+        this.recipeNoteTaker = recipeNoteTaker;
+        this.recipeManager = recipeManager;
+    }
+
+    @Override
+    public void run() {
+        reader.chooseOperation(operations, "Recipe Note Taker");
+    }
+
+    @Override
+    public String getDescription() {
+        return "Create, list, edit, and delete notes";
+    }
+
+    @Override
+    public String getCode() {
+        return "recipe notes";
+    }
+
+    private class NoteCreator implements TextualOperation {
+
+        @Override
+        public String getCode() {
+            return "create";
+        }
+
+        @Override
+        public String getDescription() {
+            return "Create a new note";
+        }
+
+        @Override
+        public void run() {
+            List<Recipe> recipes = recipeManager.getAllRecipes();
+
+            if (recipes.size() < 1) {
+                Colour.info("There are no recipes to add notes to.");
+                return;
+            }
+
+            Recipe recipe = reader.chooseFromList(recipes, "Recipe");
+
+            if (recipe == null) {
+                return;
+            }
+
+            String noteText = reader.getInput("Note Text:");
+
+            Note note = new Note(noteText, recipe);
+            recipeNoteTaker.createNote(note);
+
+            Colour.info("Created note %s", noteText);
+        }
+    }
+
+    private class NoteViewer implements TextualOperation {
+        @Override
+        public String getCode() {
+            return "view";
+        }
+
+        @Override
+        public String getDescription() {
+            return "View a recipe's notes";
+        }
+
+        @Override
+        public void run() {
+            List<Recipe> recipes = recipeManager.getAllRecipes();
+
+            Recipe recipe = reader.chooseFromList(recipes, "Recipe to View");
+
+            if (recipe != null) {
+                List<Note> notes = recipeManager.getNotes(recipe);
+
+                if (notes.size() < 1) {
+                    Colour.info("There are no notes on this recipe");
+                    return;
+                }
+
+                for (Note note : notes) {
+                    System.out.println(note);
```

Comment:
I'm not sure if this will work. I remember for my TUI operation I accidentally wrote a print statement like this instead of a Colour.info statement. I may be wrong and I unfortunately don't have the time to test this out myself, but it's something to be aware of.

Reply:
It does work.